lm_eval/models/huggingface.py

```diff
@@ -103,6 +103,7 @@ def __init__(
         use_fast_tokenizer: Optional[bool] = True,
         add_bos_token: Optional[bool] = False,
         prefix_token_id: Optional[int] = None,
+        chat_template_definition: Optional[bool | str] = None,
```

Comment:
To simplify, I would simply reuse `use_chat_template` tbh. 
It also allows you to remove these sections `model_args["chat_template_definition"] = apply_chat_template` from above, which will make the code a bit heavier to maintain.

Reply:
I think you had the `apply_chat_template` arg in mind here, right? Changed to be handled this way